### ISSUE TITLE
npm install fails

### ISSUE THREAD

**Issue by sgofferj** [eddiesigner/liebling#341]
- Downloaded master
- Unpacked
- Ran npm install in src as per instructions

```
┌─[sgofferj@enterprise][~/Dev/web/liebling-master/src]
└──▪ npm install
npm ERR! code 1
npm ERR! git dep preparation failed
npm ERR! command /usr/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/home/sgofferj/.npm/_cacache --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit
npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command sh -c node scripts/build.js
npm ERR! npm ERR! Building: /home/sgofferj/node_modules/node/bin/node /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR! npm ERR! make: Entering directory '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build'
npm ERR! npm ERR!   g++ '-DNODE_GYP_MODULE_NAME=binding' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-D__STDC_FORMAT_MACROS' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DBUILDING_NODE_EXTENSION' -I/home/sgofferj/.node-gyp/15.10.0/include/node -I/home/sgofferj/.node-gyp/15.10.0/src -I/home/sgofferj/.node-gyp/15.10.0/deps/openssl/config -I/home/sgofferj/.node-gyp/15.10.0/deps/openssl/openssl/include -I/home/sgofferj/.node-gyp/15.10.0/deps/uv/include -I/home/sgofferj/.node-gyp/15.10.0/deps/zlib -I/home/sgofferj/.node-gyp/15.10.0/deps/v8/include -I../../nan  -fPIC -pthread -Wall -Wextra -Wno-unused-parameter -m64 -O3 -fno-omit-frame-pointer -fno-rtti -fno-exceptions -std=gnu++1y -std=c++0x -MMD -MF ./Release/.deps/Release/obj.target/binding/src/binding.o.d.raw   -c -o Release/obj.target/binding/src/binding.o ../src/binding.cpp
npm ERR! npm ERR! make: Leaving directory '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build'
npm ERR! npm ERR! gyp info it worked if it ends with ok
npm ERR! npm ERR! gyp verb cli [
npm ERR! npm ERR! gyp verb cli   '/home/sgofferj/node_modules/node/bin/node',
npm ERR! npm ERR! gyp verb cli   '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/bin/node-gyp.js',
npm ERR! npm ERR! gyp verb cli   'rebuild',
npm ERR! npm ERR! gyp verb cli   '--verbose',
npm ERR! npm ERR! gyp verb cli   '--libsass_ext=',
npm ERR! npm ERR! gyp verb cli   '--libsass_cflags=',
npm ERR! npm ERR! gyp verb cli   '--libsass_ldflags=',
npm ERR! npm ERR! gyp verb cli   '--libsass_library='
npm ERR! npm ERR! gyp verb cli ]
npm ERR! npm ERR! gyp info using node-gyp@3.8.0
npm ERR! npm ERR! gyp info using node@15.10.0 | linux | x64
npm ERR! npm ERR! gyp verb command rebuild []
npm ERR! npm ERR! gyp verb command clean []
npm ERR! npm ERR! gyp verb clean removing "build" directory
npm ERR! npm ERR! gyp verb command configure []
npm ERR! npm ERR! gyp verb check python checking for Python executable "python2" in the PATH
npm ERR! npm ERR! gyp verb `which` succeeded python2 /usr/bin/python2
npm ERR! npm ERR! gyp verb check python version `/usr/bin/python2 -c "import sys; print "2.7.18
npm ERR! npm ERR! gyp verb check python version .%s.%s" % sys.version_info[:3];"` returned: %j
npm ERR! npm ERR! gyp verb get node dir no --target version specified, falling back to host node version: 15.10.0
npm ERR! npm ERR! gyp verb command install [ '15.10.0' ]
npm ERR! npm ERR! gyp verb install input version string "15.10.0"
npm ERR! npm ERR! gyp verb install installing version: 15.10.0
npm ERR! npm ERR! gyp verb install --ensure was passed, so won't reinstall if already installed
npm ERR! npm ERR! gyp verb install version is already installed, need to check "installVersion"
npm ERR! npm ERR! gyp verb got "installVersion" 9
npm ERR! npm ERR! gyp verb needs "installVersion" 9
npm ERR! npm ERR! gyp verb install version is good
npm ERR! npm ERR! gyp verb get node dir target node version installed: 15.10.0
npm ERR! npm ERR! gyp verb build dir attempting to create "build" dir: /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build
npm ERR! npm ERR! gyp verb build dir "build" dir needed to be created? /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build
npm ERR! npm ERR! gyp verb build/config.gypi creating config file
npm ERR! npm ERR! gyp verb build/config.gypi writing out config file: /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build/config.gypi
npm ERR! npm ERR! gyp verb config.gypi checking for gypi file: /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/config.gypi
npm ERR! npm ERR! gyp verb common.gypi checking for gypi file: /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/common.gypi
npm ERR! npm ERR! gyp verb gyp gyp format was not specified; forcing "make"
npm ERR! npm ERR! gyp info spawn /usr/bin/python2
npm ERR! npm ERR! gyp info spawn args [
npm ERR! npm ERR! gyp info spawn args   '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! npm ERR! gyp info spawn args   '-f',
npm ERR! npm ERR! gyp info spawn args   'make',
npm ERR! npm ERR! gyp info spawn args   '-I',
npm ERR! npm ERR! gyp info spawn args   '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass/build/config.gypi',
npm ERR! npm ERR! gyp info spawn args   '-I',
npm ERR! npm ERR! gyp info spawn args   '/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/addon.gypi',
npm ERR! npm ERR! gyp info spawn args   '-I',
npm ERR! npm ERR! gyp info spawn args   '/home/sgofferj/.node-gyp/15.10.0/include/node/common.gypi',
npm ERR! npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/sgofferj/.node-gyp/15.10.0',
npm ERR! npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp',
npm ERR! npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/sgofferj/.node-gyp/15.10.0/<(target_arch)/node.lib',
npm ERR! npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass',
npm ERR! npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! npm ERR! gyp info spawn args   '--depth=.',
npm ERR! npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! npm ERR! gyp info spawn args   '--generator-output',
npm ERR! npm ERR! gyp info spawn args   'build',
npm ERR! npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! npm ERR! gyp info spawn args ]
npm ERR! npm ERR! gyp verb command build []
npm ERR! npm ERR! gyp verb build type Release
npm ERR! npm ERR! gyp verb architecture x64
npm ERR! npm ERR! gyp verb node dev dir /home/sgofferj/.node-gyp/15.10.0
npm ERR! npm ERR! gyp verb `which` succeeded for `make` /usr/bin/make
npm ERR! npm ERR! gyp info spawn make
npm ERR! npm ERR! gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! npm ERR! In file included from ../src/binding.cpp:3:
npm ERR! npm ERR! ../src/sass_context_wrapper.h:8:10: fatal error: sass/context.h: No such file or directory
npm ERR! npm ERR!     8 | #include <sass/context.h>
npm ERR! npm ERR!       |          ^~~~~~~~~~~~~~~~
npm ERR! npm ERR! compilation terminated.
npm ERR! npm ERR! make: *** [binding.target.mk:126: Release/obj.target/binding/src/binding.o] Error 1
npm ERR! npm ERR! gyp ERR! build error 
npm ERR! npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/lib/build.js:262:23)
npm ERR! npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:378:20)
npm ERR! npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
npm ERR! npm ERR! gyp ERR! System Linux 5.4.0-67-generic
npm ERR! npm ERR! gyp ERR! command "/home/sgofferj/node_modules/node/bin/node" "/home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
npm ERR! npm ERR! gyp ERR! cwd /home/sgofferj/.npm/_cacache/tmp/git-clone-d5dba9ce/node_modules/node-sass
npm ERR! npm ERR! gyp ERR! node -v v15.10.0
npm ERR! npm ERR! gyp ERR! node-gyp -v v3.8.0
npm ERR! npm ERR! gyp ERR! not ok 
npm ERR! npm ERR! Build failed with error code: 1
npm ERR! 
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     /home/sgofferj/.npm/_cacache/_logs/2021-03-18T12_11_51_515Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sgofferj/.npm/_logs/2021-03-18T12_11_52_953Z-debug.log
```

**Comment by eddiesigner** [comment]
What OS are you using? Node version?

You could run `npm rebuild node-sass` and try again. You can also remove the `node_modules` folder and install the dependencies with `LIBSASS_EXT="no" npm install` You might need also to install `libsass ` on your machine.

**Comment by sgofferj** [comment]
I originally had node 14.10, tried updating to 15.12 but that didn't help.
With `LIBSASS_EXT="no" npm install` npm install goes through now. 
npm run watch doesn't seem work anymore, there's nothing listening on 2368.
npm run docker-watch works.

**Comment by sgofferj** [comment]
npm run production does not produce a zip file.
```
 DONE  Compiled successfully in 4666ms                                                                                                              11:14:17

                        Asset       Size  Chunks             Chunk Names
   /fonts/icomoon/icomoon.eot   14.3 KiB          [emitted]  
   /fonts/icomoon/icomoon.svg   50.7 KiB          [emitted]  
   /fonts/icomoon/icomoon.ttf   14.1 KiB          [emitted]  
  /fonts/icomoon/icomoon.woff   14.2 KiB          [emitted]  
                   /js/app.js   6.78 KiB       2  [emitted]  /js/app
               /js/helpers.js   2.01 KiB       3  [emitted]  /js/helpers
                  /js/home.js   3.27 KiB       4  [emitted]  /js/home
              /js/manifest.js   1.46 KiB       0  [emitted]  /js/manifest
                  /js/page.js   2.08 KiB       5  [emitted]  /js/page
                  /js/post.js   4.18 KiB       6  [emitted]  /js/post
                /js/vendor.js    202 KiB       1  [emitted]  /js/vendor
/js/vendor/content-api.min.js   36.4 KiB          [emitted]  
                  css/404.css   1.06 KiB       3  [emitted]  /js/helpers
                  css/app.css     63 KiB       3  [emitted]  /js/helpers
                 css/home.css   3.88 KiB       3  [emitted]  /js/helpers
              css/listing.css  576 bytes       3  [emitted]  /js/helpers
           css/newsletter.css   2.82 KiB       3  [emitted]  /js/helpers
                 css/post.css   21.7 KiB       3  [emitted]  /js/helpers
                 css/tags.css   1.47 KiB       3  [emitted]  /js/helpers

> liebling@1.0.0 zip
> run-script-os

This is meant to be run from within npm script. See https://github.com/charlesguse/run-script-os
```

**Comment by clitzinger** [comment]
Hi !

I had the same issue and tried many stuffs on Windows (Python, build tools, node-sass version, etc.).
At the end, i found that this dependency was responsible : "aos": "eddiesigner/aos.git#v2".
My solution was to get the official latest version of "aos" :

- npm remove aos
- npm install aos
- npm install

Working fine for me, i hope this help!
@eddiesigner You can use "^2.3.4" version (i didn't saw regression, but you are the expert 👍 )

**Comment by eddiesigner** [comment]
@clitzinger Thanks for your input! However I don't see how that can be related 🤔  How did you find that this dependency was the culprit?

I made a fork of AOS because the original implemented a scroll listener without the `passive` flag which might cause some minimal performance issues. In the next version I will use the original one and maybe in the future I will use another library but I'm not sure yet.

**Comment by clitzinger** [comment]
Hi @eddiesigner,

Sorry for the response delay.
I found out the hard way, trying for a moment to circumvent the problem with build-tools and python version : no result.
Then, I created an empty node project and add dependencies (dev and prod) one by one until it broke on the "aos" one 🧐.

By the way, thanks for the Ghost 4 support 👍 !
I will try in the next weeks to add pull requests for some adjustments I've done on my side (I hope they will be useful).
